src/server/ftpserver.rs

```diff
@@ -386,13 +469,32 @@ where
     /// use unftp_sbe_fs::ServerExt;
     ///
     /// let server = Server::with_fs("/tmp")
-    ///              .passive_host("ftp.myserver.org");
+    ///              .passive_host("ftp.myserver.org")
+    ///              .build()
+    ///              .unwrap();
     /// ```
     pub fn passive_host<H: Into<PassiveHost>>(mut self, host_option: H) -> Self {
         self.passive_host = host_option.into();
         self
     }
 
+    /// Preallocate a listener socket for the PASV command
+    ///
+    /// Preallocate a socket suitable for servicing the PASV command as received from the given IP
+    /// address.  This method is completely optional, but it is necessary in order to use the
+    /// [`Server::service`] method when the server is running in capability mode.
+    // In the future, this could be extended to accept multiple invocations, allocating a
+    // collection of sockets.
+    pub async fn pasv_listener(self, addr: std::net::IpAddr) -> Self {
+        // XXX: maybe should convert this into a synchronous function by using
+        // std::net::TcpListener::bind instead of tokio::net::TcpListener::bind
+        let sock = crate::server::controlchan::commands::Pasv::try_port_range(addr, self.passive_ports.clone())
```

Comment:
This logic should probably happen on `build()`

Reply:
Good point.  We can just store the IpAddr here, and open the socket during `build`.  I'll make that change.